hypertrace-graphql-gateway-service-metric-utils/src/main/java/org/hypertrace/graphql/utils/metrics/gateway/BaselineIntervalListContainer.java

```diff
@@ -0,0 +1,180 @@
+package org.hypertrace.graphql.utils.metrics.gateway;
+
+import io.reactivex.rxjava3.core.Observable;
+import io.reactivex.rxjava3.core.Single;
+import io.reactivex.rxjava3.observables.GroupedObservable;
+import lombok.Getter;
+import lombok.Value;
+import lombok.experimental.Accessors;
+import org.hypertrace.core.graphql.common.utils.CollectorUtils;
+import org.hypertrace.core.graphql.common.utils.Converter;
+import org.hypertrace.gateway.service.v1.baseline.Baseline;
+import org.hypertrace.gateway.service.v1.baseline.BaselineInterval;
+import org.hypertrace.gateway.service.v1.baseline.BaselineMetricSeries;
+import org.hypertrace.graphql.metric.request.MetricAggregationRequest;
+import org.hypertrace.graphql.metric.request.MetricSeriesRequest;
+import org.hypertrace.graphql.metric.schema.BaselineMetricInterval;
+import org.hypertrace.graphql.metric.schema.BaselinedMetricAggregation;
+import org.hypertrace.graphql.metric.schema.Health;
+import org.hypertrace.graphql.metric.schema.MetricBaselineAggregation;
+
+import javax.inject.Inject;
+import java.time.Instant;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+class BaselineIntervalListContainer
+    implements Converter<
+        Map<MetricSeriesRequest, BaselineMetricSeries>, List<BaselineMetricInterval>> {
+
+  private final MetricBaselineAggregationConverter baselineAggregationConverter;
+
+  @Inject
+  BaselineIntervalListContainer(MetricBaselineAggregationConverter baselineAggregationConverter) {
```

Comment:
this looks like mostly copy paste from the existing interval list converter - how much of the logic could be shared? I expect most of it could go in a base class that is just parameterized with the specific impementation builders.